### DISCUSSION TITLE
Difference in JIT and AOT schedulling

### DISCUSSION THREAD

**Discussion by Doubl3MD** [halide/Halide#7199] [Q&A]
With this schedule :

```Var x_outer, y_outer, x_inner, y_inner, tile_index;
   main
      .tile(x, y, x_outer, y_outer, x_inner, y_inner, 64, 64)
      .fuse(x_outer, y_outer, tile_index)
      .parallel(tile_index);
   Var x_inner_outer, y_inner_outer, x_vectors, y_pairs;
   main
      .tile(x_inner, y_inner, x_inner_outer, y_inner_outer, x_vectors, y_pairs, 4, 2)
      .vectorize(x_vectors)
      .unroll(y_pairs);
```
When I JIT compile for testing over an image with size smaller than the tile size of 64,64 it passes.
When I AOT and use the function call on the same image I get an error(crash and burn). 

It's logic, I guess, that it doesn't work when the tile size is bigger than the full image size, but I would expect that with JIT it would produce a similar error, which it doesn't.

I'm just looking for the logical explanation why or why not...  so, anybody? :)

**Comment by abadams** [comment]
JIT mode checks to see if the output is too small, and if so computes a bigger output and returns a crop of it.

**Reply by steven-johnson** [reply]
I suspect that using JIT mode with `compile_to_callable()`, rather than `realize()`, would likely mimic AOT behavior exactly.

**Reply by soufianekhiat** [reply]
Reference for similar issue:
https://github.com/halide/Halide/issues/5915
